File: 6__Python__/pset6/credit/credit.py
AMEX = ["34", "37"]
MASTERCARD = [str(i) for i in range(51, 56)]
VISA = ["4"]

number = "a"
while not number.isdigit():
    number = input("Number: ")
card_type = None
if number[:2] in AMEX:
    card_type = "AMEX"
elif number[:2] in MASTERCARD:
    card_type = "MASTERCARD"
elif number[0] in VISA:
    card_type = "VISA"

if card_type:
    number = int(number)
    total = 0
    mult = False
    while number:
        digit = (number % 10) * (1 + int(mult))
        while digit:
            total += digit % 10
            digit //= 10
        number //= 10
        mult = not mult
    if total % 10 == 0:
        print(card_type)
    else:
        print("INVALID")

else:
    print("INVALID")

File: 1__C__/pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void){
    // AMEX 34, 37
    // MASTERCARD 51, 52, 53, 53, 54, 55
    // VISA 4
    long num = get_long("Number: ");
    int total = 0;
    string type = "INVALID\n";
    while (num > 0)
    {
        if (num < 100)
        {
            if (num == 34 || num == 37)
            {
                type = "AMEX\n";
            }
            else if (num > 50 && num < 56)
            {
                type = "MASTERCARD\n";
            }
            else if (num == 4 || num / 10 == 4)
            {
                type = "VISA\n";
            }
        }
        total += num % 10;
        num /= 10;
        if (num < 100)
        {
            if (num == 34 || num == 37)
            {
                type = "AMEX\n";
            }
            else if (num > 50 && num < 56)
            {
                type = "MASTERCARD\n";
            }
            else if (num == 4 || num/10 == 4)
            {
                type = "VISA\n";
            }
        }
        int curr = (num % 10) * 2;
        while (curr > 0)
        {
            total += curr % 10;
            curr /= 10;
        }
        num /= 10;
    }
    if (total % 10 == 0)
    {
        printf("%s", type);
    }
    else
    {
        printf("INVALID\n");
    }
}